translation-web/src/main/java/eu/europeana/api/translation/web/TranslationController.java

```diff
@@ -39,8 +46,14 @@ public ResponseEntity<String> translate(@RequestBody TranslationRequest translRe
     verifyWriteAccess(Operations.CREATE, request);
 
     validateRequest(translRequest);
-
-    TranslationResponse result = translationService.translate(translRequest);
+    
+    TranslationResponse result = null;
+    if(translRequest.isCaching()) {
+      result = getCombinedCachedAndTranslatedResults(translRequest);
```

Comment:
this method belongs to the business logic threrefore it needs to be moved to the TranslationWebService and called within the translate method